app/src/main/java/ai/elimu/analytics/task/UploadEventsWorker.kt

```diff
@@ -37,14 +38,16 @@ class UploadEventsWorker(context: Context, workerParams: WorkerParameters) :
 
         // Upload CSV files to the server
         // Example format:
-        //   files/version-code-3001012/letter-assessment-events/7161a85a0e4751cd_3001012_letter-assessment-events_2020-03-21.csv
+        //   files/lang-HIN/letter-assessment-events/7161a85a0e4751cd_3003002_letter-assessment-events_2025-06-07.csv
         val filesDir = applicationContext.filesDir
-        for (versionCodeDir in filesDir.listFiles() ?: emptyArray()) {
-            Timber.i("versionCodeDir: $versionCodeDir")
-            if (versionCodeDir.name.startsWith("version-code-")) {
-                val wordLearningEventsDir = File(versionCodeDir, eventType.type)
-                Timber.i("Uploading CSV files from $wordLearningEventsDir")
-                val files = wordLearningEventsDir.listFiles()
+        val language = SharedPreferencesHelper.getLanguage(applicationContext)
+        Timber.i("language: ${language}")
+        for (file in filesDir.listFiles() ?: emptyArray()) {
+            Timber.i("file.name: ${file.name}")
+            if (file.name.startsWith("lang-${language}")) {
```

Comment:
There should be an extra check if log file matches upload even type before uploading.
Otherwise, a log file will be uploaded via API endpoint which should be used to upload different type of log file.
Refer to https://github.com/elimu-ai/analytics/pull/295 in file app/src/main/java/ai/elimu/analytics/task/UploadEventsWorker.kt line 43.

Reply:
@tuancoltech Are you referring to CSV files within the `lang-<LANGUAGE>` directory?

After the changes in this PR, the directory structure will look like this:

![image](https://github.com/user-attachments/assets/f1ad923e-ed80-455c-aa63-55f7009b7ebb)

So there won't be any CSV files inside the language directory itself, only directories.

Reply:
Oh yes, I see.